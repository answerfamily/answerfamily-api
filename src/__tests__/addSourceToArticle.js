jest.mock('../dataloaders/auth0UserLoaderFactory');

const {
  loadESFixtures,
  unloadESFixtures,
  loadMongoFixtures,
  unloadMongoFixtures,
} = require('../../test/fixtures');
const gql = require('../../test/gql');
const {
  ES_FIXTURES,
  MONGO_FIXTURES,
} = require('../__fixtures__/articleParagraphReply');
const mongoClient = require('../lib/mongoClient');

describe('addSourceToArticle', () => {
  beforeAll(async () => {
    await loadESFixtures(ES_FIXTURES);
    await loadMongoFixtures(MONGO_FIXTURES);
  });

  afterAll(async () => {
    await unloadESFixtures(ES_FIXTURES);
    await unloadMongoFixtures(MONGO_FIXTURES);
  });

  const source = {
    url: 'http://unittest.com',
    note: 'description',
  };
  it('creates sources', async () => {
    const { errors, data } = await gql`
      mutation($articleId: String!, $source: ArticleSourceInput!) {
        addSourceToArticle(articleId: $articleId, source: $source) {
          sources {
            note
            url
          }
        }
      }
    `(
      { articleId: 'article1', source },
      {
        userPromise: Promise.resolve({ iss: 'user-id' }),
      }
    );

    expect(errors).toBeUndefined();
    expect(data.addSourceToArticle.sources).toMatchSnapshot();

    // cleanup
    const { db } = await mongoClient;
    await db.collection('sources').deleteMany({ url: source.url });
  });

  it('blocks non-logged in users', async () => {
    const { errors } = await gql`
      mutation($articleId: String!, $source: ArticleSourceInput!) {
        addSourceToArticle(articleId: $articleId, source: $source) {
          id
        }
      }
    `({ articleId: 'article1', source });

    // Should be unauthorzied
    expect(errors).toMatchSnapshot();
  });
});
