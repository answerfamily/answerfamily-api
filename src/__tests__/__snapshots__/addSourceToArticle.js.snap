// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addSourceToArticle blocks non-logged in users 1`] = `
Array [
  [GraphQLError: Must log in first],
]
`;

exports[`addSourceToArticle creates sources 1`] = `
Array [
  Object {
    "note": null,
    "url": "http://google.com",
  },
  Object {
    "note": "description",
    "url": "http://unittest.com",
  },
]
`;

exports[`addSourceToArticle creates sources: scrapUrl calls 1`] = `
Array [
  Array [
    "http://unittest.com",
    Object {
      "cacheLoader": DataLoader {
        "_batchLoadFn": [Function],
        "_options": undefined,
        "_promiseCache": Map {},
        "_queue": Array [],
      },
      "client": Promise {},
      "persist": true,
    },
  ],
]
`;
